src/main/ffmpeg/ffmpeg.ts

```diff
@@ -0,0 +1,147 @@
+import { exec as _exec } from 'child_process';
+import decompress from 'decompress';
+import log from 'electron-log';
+import fs from 'fs/promises';
+import Downloader from 'nodejs-file-downloader';
+import os from 'os';
+import path from 'path';
+import util from 'util';
+import { FfmpegInfo } from '../../common/common-types';
+import { isM1 } from '../../common/env';
+import { assertNever, checkFileExists } from '../../common/util';
+
+const exec = util.promisify(_exec);
+
+const downloads: Record<SupportedPlatform, string> = {
+    linux: 'https://github.com/chaiNNer-org/ffmpeg-rehost/releases/download/ffmpeg/ffmpeg-linux-x64.zip',
+    darwin: isM1
+        ? 'https://github.com/chaiNNer-org/ffmpeg-rehost/releases/download/ffmpeg/ffmpeg-darwin-arm64.zip'
+        : 'https://github.com/chaiNNer-org/ffmpeg-rehost/releases/download/ffmpeg/ffmpeg-darwin-x64.zip',
+    win32: 'https://github.com/chaiNNer-org/ffmpeg-rehost/releases/download/ffmpeg/ffmpeg-win32-x64.zip',
+};
+
+type SupportedPlatform = 'linux' | 'darwin' | 'win32';
+
+const getPlatform = (): SupportedPlatform => {
+    const platform = os.platform();
+    switch (platform) {
+        case 'win32':
+        case 'linux':
+        case 'darwin':
+            return platform;
+        default:
+            throw new Error(
+                `Unsupported platform: Our FFMPEG is not supported on ${platform}. Please report this to us and we may add support.`
+            );
+    }
+};
```

Comment:
Pretty sure I used this somewhere else too. Maybe we could move this in a `src/main/platform.ts`?

Reply:
That's a good idea. And yeah a lot of this stuff is similar to the python downloading stuff. If we ever need a third thing like this we should probably abstract it away a bit more